archive_converter.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import email
+import os
+from setup_gmail import Message
+
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
+        self._messages_seen = {} # Maps message.id to message
+
+    def find_thread(self, key: str) -> Message:
+        return self._lookup_index[key]
+
+    def update(self):
+        fully_updated = False
+        new_messages = []
+
+        for filename in os.listdir('test_data'):
+            if filename.endswith(".txt"):
+                email = generate_email_from_file('test_data/' + filename)
+                if(email.id not in self._messages_seen):
+                    new_messages.append(email)
+        self._add_messages_to_index(new_messages)
+
+    def _add_messages_to_index(self, new_messages):
+        need_parent = []
+        for message in new_messages:
+            self._messages_seen[message.id] = message
+
+            if message.in_reply_to is None:
+                key = message.get_key()
+                print(key)
+                if key:
+                    self._lookup_index[key] = message
+                else:
+                    # If message doesn't have the LKML style [PATCH v4 00/11],
+                    # we don't need to put it in the index.
+                    continue
```

Comment:
Ditto, put this continue earlier

```py
if key is None:
    # comment
    continue
...
```

I'd also flip the outer conditional as well

```py
if messsage.in_reply_to is not None:
     need_parent.append(message)
     continue
...
```